pkg/build/controller/build/build_controller.go

```diff
@@ -425,6 +597,22 @@ func (bc *BuildController) createBuildPod(build *buildapi.Build) (*buildUpdate,
 		return update, err
 	}
 
+	// image reference resolution requires a copy of the build
+	build, err = buildutil.BuildDeepCopy(build)
+	if err != nil {
+		return nil, fmt.Errorf("unable to copy build %s: %v", buildDesc(build), err)
+	}
+
+	// Set the output Docker image reference.
+	if err := bc.resolveImageReferences(build); err != nil {
+		// If we cannot resolve the output reference, the output image stream
+		// may not yet exist. The build should remain in the new state and show the
+		// reason that it is still in the new state.
+		update.setReason(buildapi.StatusReasonInvalidImageReference)
+		update.setMessage(buildapi.StatusMessageInvalidImageRef)
+		return update, err
```

Comment:
@bparees this pr uses a [resourceTriggerQueue](https://github.com/openshift/origin/pull/15156/files#diff-4349d2367514c13a1d5c76dc4debf58fR49) to keep track of imagestreams that are referenced by a build for which an imagestreamtag or imagestreamimage [failed to resolve](https://github.com/openshift/origin/pull/15156/files#diff-4349d2367514c13a1d5c76dc4debf58fR522). When an [add/update happens](https://github.com/openshift/origin/pull/15156/files#diff-4349d2367514c13a1d5c76dc4debf58fR937) on that tracked imagestream, then the corresponding builds are [queued](https://github.com/openshift/origin/pull/15156/files#diff-4349d2367514c13a1d5c76dc4debf58fR938) again.

Which is why I'm saying that the build doesn't need to be retried :)